osu.Game/Screens/Backgrounds/BackgroundScreenDefault.cs

```diff
@@ -86,5 +161,22 @@ private void load()
                 Sprite.Texture = skin.GetTexture("menu-background") ?? Sprite.Texture;
             }
         }
+
+        private class BeatmapBackground : Background
+        {
+            private readonly WorkingBeatmap beatmap;
+
+            public BeatmapBackground(WorkingBeatmap beatmap, string fallbackTextureName)
```

Comment:
fallback is not going to work as you expect. check the base code - it's *always* going to load it.